### ISSUE TITLE
Where the checkpoints are saved

### ISSUE THREAD

**Issue by mcabinaya** [jiny2001/dcscn-super-resolution#31]
I am running the code for DIV2K dataset with all default parameters. It has run for more than 25 epochs. I don't see model checkpoints getting saved anywhere. Is there something wrong?
How do I reload the model after certain epoch and evaluate on test data?

**Comment by jiny2001** [comment]
Oops, it doesn't save on the way. It only saves the model when the whole training finished.
Once I saved on every epoch but in that case I found my checkpoint file get really huge. 

It'd be better save the model in every 2 or 3 epochs and keeps only latest 4 to 5 epochs by default?

Will try to change it on this weekend. Thx.

**Comment by mcabinaya** [comment]
Yes Thank you @jiny2001 
That would be helpful so that we can look into the training and validation loss to stop the training and avoid overfitting.

**Comment by jiny2001** [comment]
Hi,
I updated the project. In the latest branch, model will be saved when the performance get improved.
At the end of the every epoch, we check if the PSNR for the test dataset is improved or not. And save the model if it is improved.

The other way is keeping every (or some latest) models. In that case, specify "max_to_keep=" to a big number in tf.train.Saver() at build_summary_saver() in helper/tf_graph.py. And then specify "global_step=" to epoch number at save_model().

I tested that way but I found it doesn't match my style. So I decided to keep current style, just keep one best performance model and update at the last. Because I believe it won't overfit in most cases.

When it overfits, I think we would better try the other generalization technic, not early stopping. Feel free to ask if you have any suggestion or you opinion, thx!!
